cmd/clusterctl/pkg/client/repository/template.go

```diff
@@ -45,9 +47,85 @@ type Template interface {
 	Variables() []string
 
 	// TargetNamespace where the template objects will be installed.
-	// This value is inherited by the template options.
 	TargetNamespace() string
 
-	// Yaml file defining all the cluster objects.
-	Yaml() []byte
+	// Yaml file defining all the cluster template objects.
+	Yaml() ([]byte, error)
+
+	// Objs return the cluster template  in the form of a list of Unstructured objects.
+	Objs() []unstructured.Unstructured
+}
+
+// template implement Template
+type template struct {
+	config.Provider
+	version         string
+	flavor          string
+	bootstrap       string
+	variables       []string
+	targetNamespace string
+	objs            []unstructured.Unstructured
+}
+
+// Ensures template implement the Template interface.
+var _ Template = &template{}
+
+func (t *template) Version() string {
+	return t.version
+}
+
+func (t *template) Flavor() string {
+	return t.flavor
+}
+
+func (t *template) Bootstrap() string {
+	return t.bootstrap
+}
+
+func (t *template) Variables() []string {
+	return t.variables
+}
+
+func (t *template) TargetNamespace() string {
+	return t.targetNamespace
+}
+
+func (t *template) Objs() []unstructured.Unstructured {
+	return t.objs
+}
+
+func (t *template) Yaml() ([]byte, error) {
+	return util.FromUnstructured(t.objs)
+}
+
+// newTemplate returns a new objects embedding a cluster template YAML file.
+func newTemplate(provider config.Provider, version, flavor, bootstrap string, rawyaml []byte, configVariablesClient config.VariablesClient, targetNamespace string) (*template, error) {
```

Comment:
```suggestion
func newTemplate(provider config.Provider, version, flavor, bootstrap string, rawYaml []byte, configVariablesClient config.VariablesClient, targetNamespace string) (*template, error) {
```

Reply:
This has enough arguments that we should probably convert them into an input struct

Reply:
done